### PR TITLE
terramate: init at 0.3.0

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

https://github.com/terramate-io/terramate

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/terramate/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "terramate-io";
+    repo = "terramate";
+    rev = "v${version}";
+    hash = "sha256-/e3NbUhHQlREjb3YYSfrPq/GE+BfFI3mUO4eRE4mk9U=";
+  };
+
+  vendorHash = "sha256-bja4v48BSPZC/vJVps2LKq90fReYpMY/yKTThsQQCHE=";
+
+  nativeBuildInputs = [ git ];
```
Comment (NickCao) [inline pkgs/development/tools/misc/terramate/default.nix:20]: What's git for?

File: pkgs/development/tools/misc/terramate/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  ldflags = [ "-extldflags" "-static" ];
```
Comment (NickCao) [inline pkgs/development/tools/misc/terramate/default.nix:22]: Linking it as a dynamic executable is fine.

File: pkgs/development/tools/misc/terramate/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  meta = with lib; {
+    description = "Maintain highly scalable Terraform projects by adding powerful capabilities such as code generation, stacks, orchestration, change detection, data sharing and more";
```
Comment (NickCao) [inline pkgs/development/tools/misc/terramate/default.nix:25]: Keep the description short while informative.

File: pkgs/development/tools/misc/terramate/default.nix (hunk 1 above)
Comment (dit7ya) [reply pkgs/development/tools/misc/terramate/default.nix:21]: Something related to version. Removing it requires extra patching.

**Review by NickCao — APPROVED** [review]

**Pull request merged**